contracts/rdt/RevenueDistributionToken.sol

```diff
@@ -0,0 +1,414 @@
+    }
+
+    function _claim(address receiver_) internal returns (uint256 claimableRevenueAssets_) {
+        claimableRevenueAssets_ = claimableRevenueAssets(receiver_);
+        require(ERC20Helper.transfer(revenueAsset, receiver_, claimableRevenueAssets_), "RDT:C:TRANSFER");
+
+        lastClaimed[receiver_] = block.number;
+        _updateIssuanceParams();
+
+        emit Claim(receiver_, claimableRevenueAssets_);
+    }
+
+    function _updateIssuanceParams() internal returns (uint256 issuanceRate_) {
+        return issuanceRate = (lastUpdated = block.number) > vestingPeriodFinish ? 0 : issuanceRate;
+    }
+
+    function claimAndStake(
+        address masterChef_,
+        address receiver_,
+        uint256 pid_
+    ) external {
+        uint256 claimedRevenueAssets_ = _claim(receiver_);
+
+        ERC20(revenueAsset).approve(masterChef_, claimedRevenueAssets_);
+        IMasterChefV2(masterChef_).deposit(pid_, claimedRevenueAssets_, receiver_);
+    }
+
+    /**********************/
+    /*** View Functions ***/
+    /**********************/
+
+    function claimableRevenueAssets(address account_) public view virtual returns (uint256 balanceOfAssets_) {
+        uint256 issuanceRate_ = issuanceRate;
+
+        if (ERC20(revenueAsset).balanceOf(address(this)) == 0) return 0;
+        if (issuanceRate_ == 0) return freeAssets;
+
+        uint256 supply = totalSupply; // Cache to stack.
+        uint256 shares_ = balanceOf[account_];
+
+        if (supply == 0 || shares_ == 0) return 0;
+
+        uint256 vestingPeriodFinish_ = vestingPeriodFinish;
+
+        uint256 vestingTimePassed = block.number > vestingPeriodFinish_
+            ? vestingPeriodFinish_ - lastClaimed[account_] // block deposits if vesting perdio finished
+            : block.number - lastClaimed[account_];
+
+        balanceOfAssets_ = _divRoundUp((shares_ * ((issuanceRate_ * vestingTimePassed))), supply * precision);
+
+        uint256 revenueAssetBalance_ = ERC20(revenueAsset).balanceOf(address(this));
+
+        return balanceOfAssets_ > revenueAssetBalance_ ? revenueAssetBalance_ : balanceOfAssets_;
+    }
+
+    function convertToShares(uint256 assets_) public view virtual override returns (uint256 shares_) {
+        return assets_;
+    }
+
+    function maxDeposit(address receiver_) external pure virtual override returns (uint256 maxAssets_) {
+        receiver_; // Silence warning
+        maxAssets_ = type(uint256).max;
+    }
+
+    function maxMint(address receiver_) external pure virtual override returns (uint256 maxShares_) {
+        receiver_; // Silence warning
+        maxShares_ = type(uint256).max;
+    }
+
+    function maxRedeem(address owner_) external pure virtual override returns (uint256 maxShares_) {
+        owner_; // Silence warning
+        maxShares_ = type(uint256).max;
+    }
+
+    function maxWithdraw(address owner_) external view virtual override returns (uint256 maxAssets_) {
+        owner_; // Silence warning
```

Comment:
why do we have this variable in here? is it to silence warning?
if the variable is not being used can we remove it?

Reply:
I just see that its mentioned in the EIP: https://eips.ethereum.org/EIPS/eip-4626

can we do something like
```
function maxWithdraw(address owner_)
```
or that will break the EIP?